recipes/onnxruntime/all/conanfile.py

```diff
@@ -64,57 +46,53 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
-        # onnxruntime forces this to be True
-        # https://github.com/microsoft/onnxruntime/blob/be76e1e1b8e2914e448d12a0cc683c00014c0490/cmake/external/onnxruntime_external_deps.cmake#L542
-        self.options["onnx"].disable_static_registration = True
 
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
         required_onnx_version = self.conan_data["onnx_version_map"][self.version]
         self.requires(f"onnx/{required_onnx_version}")
-        self.requires("abseil/20240116.1")
-        self.requires("protobuf/3.21.12")
-        self.requires("date/3.0.1")
-        self.requires("re2/20231101")
+        self.requires("abseil/[>=20240116.1 <=20250814.0]")
+        self.requires("protobuf/[>=3.21.12 <7]")
+        self.requires("date/[>=3.0.1 <3.1]")
+        self.requires("re2/[>=20231101]")
         self.requires("flatbuffers/23.5.26")
-        # using 1.84.0+ fails on CCI as it prevents the cpp 17 version to be picked up when building with cpp 20
         self.requires("boost/1.83.0", headers=True, libs=False)  # for mp11, header only, no need for libraries
```

Comment:
```suggestion
        self.requires("boost/[>=1.83.0 <1.90.0]", headers=True, libs=False)  # for mp11, header only, no need for libraries
```

Tested locally, it should work with Boost 1.89.0 as well: 

[onnxruntime-1.23.2-with-boost.log](https://github.com/user-attachments/files/24053997/onnxruntime-1.23.2-with-boost.log)